### PR TITLE
Accordion: multiple fixes.

### PR DESCRIPTION
**Pull request by brunano21**
This PR description:
- It fixes failing tests, caused by changes introduced by a156e55 (#671).
- It fixes keyboard navigation within a custom Accordion header (#672).

### CODE REVIEW

**Comment by brunano21** [comment]
Regarding commit #e16235c, I see that the navigation within a custom Accordion header is not the only thing broken. In fact, List is affected by the same problem, because the function at this [line](https://github.com/ibm-js/deliteful/blob/master/list/List.js#L1035) is never called, since KeyNav does not fire `delite-deactivated` event anymore. Digging more, I see that recent [KeyNav changes](https://github.com/ibm-js/delite/commit/3b7c96f074e898269f6a08f23c45c5c3329189b8) may be the reason of all these problems. 

@wkeese what do you think about?

**Comment by wkeese** [comment]
I see the navigation within a custom Accordion header is working on Chrome, but not on IE11.

On IE11, when the AccordionHeader is focused, there's a focus event on the AccordionHeader (as expected) so it gets a `delite-activated` event.   So far so good.  But then there's immediately another focus event on the Accordion itself.   Thus the AccordionHeader gets a `delite-deactivated` event, and the `<button>` inside the AccordionHeader is set to tabIndex -1, so it's not tab navigable.

I'm not sure yet why the Accordion gets a focus event.  `Accordion#hasAttribute("tabindex")` is false.

**Comment by wkeese** [comment]
OK, the problem is the order methods are called:

1. User tabs to Accordion.   This will cause a `focusin` event and then a `focus` event.
2. Accordion `focusin` event triggers KeyNav#focusinHandler().
4. KeyNav#focusinHandler() calls KeyNav#focus() which eventually calls AccordionHeader#focus().
5. AccordionHeader gets `focus` event which makes activationTracker send `delite-activated` event to AccordionHeader.
6. Accordion `focusin` event processing finally finished.
7 Accordion `focus` event processing starts.    That makes activationTracker send `delite-deactivated` event to AccordionHeader because it thinks focus has shifted back to the Accordion itself.

TL;DR activationTracker gets confused when focus changes inside a `focusin` handler.

I will work on a fix for that.

**Comment by wkeese** [comment]
I pushed these changes after updating the checkin comments to be more descriptive, and also pushed a delite fix for the problem listed above (ibm-js/delite@ e232bc563e5578b79577532921cf4ee1bf04c0ce).
